test/OpenTelemetry.Exporter.Geneva.Tests/TestHelpers.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.IO;
+
+namespace OpenTelemetry.Exporter.Geneva.Tests;
+
+internal static class TestHelpers
+{
+    internal static string GenerateTempFilePath()
+    {
+        while (true)
```

Comment:
Would it be better if the path is following some pattern - e.g. Base Path + Name of the Test Method + Timestamp + Random Number? Then there is no need to have a while(true) loop.

Reply:
Good point, but I think we don't need this while(true). We are already doing `Path.GetRandomFileName()`.

Reply:
> Good point, but I think we don't need this while(true). We are already doing `Path.GetRandomFileName()`.

I don't understand this, would you elaborate?

Reply:
So what we are doing [here](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/de2d61e3c52a2cb8164399bec0b05004264596a2/test/OpenTelemetry.Exporter.Geneva.Tests/TestHelpers.cs#L15-L18) is adding an extra layer of validation whether the Path returned by already exists or not. However, Its highly unlikely that it would, because

1) GetRandomFileName returns `cryptographically strong random 8.3 string` https://github.com/dotnet/runtime/blob/f1c5823c98a7e8612621d253a1291df2e515eb07/src/libraries/System.Private.CoreLib/src/System/IO/Path.cs#L275-L278 so there are very low chances of getting same value twice.
2) We delete the file after the use in tests.

so based on this I think it safe to remove this while(true) without adding custom logic for generating file name.

Reply:
> 1. GetRandomFileName returns `cryptographically strong random 8.3 string` https://github.com/dotnet/runtime/blob/f1c5823c98a7e8612621d253a1291df2e515eb07/src/libraries/System.Private.CoreLib/src/System/IO/Path.cs#L275-L278 so there are very low chances of getting same value twice.

That seems to a tool for a different purpose. Here we don't need a cryptographically strong random name, we don't need the name to be random at all. What we need is a unique name, exactly as what you did in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/0819114666abfdce08d05f879fda5ac4e071ef7f/src/OpenTelemetry.PersistentStorage.FileSystem/PersistentStorageHelper.cs#L125.

Reply:
Or let me use a different way - why would we took `GetUniqueFileName` approach in the persistent storage while here we use random name?

Reply:
I see your point. That makes sense, I will update.